lib/bundler/plugin/index.rb

```diff
@@ -24,6 +24,7 @@ def initialize
         @plugin_paths = {}
         @commands = {}
         @sources = {}
+        @hooks = {}
```

Comment:
Actually I did that and then reversed it (ebfebc5).

It was leading to empty hashes in the serialized index file for the hooks that are not registered but were being queried .